test/sharness/lib/test-lib.sh

```diff
@@ -9,17 +9,15 @@
 # use the ipfs tool to test against
 
 # add current directory to path, for ipfs tool.
-if test "$MAKE_SKIP_PATH" != "1"; then
```

Comment:
What changed with `MAKE_SKIP_PATH`?

Reply:
I'm not sure exactly why this fixes it. Presumably something somewhere else has a side efffect of setting the PATH so that it contains the correct ipfs binary for the test, which is what the sharness tests were implicitly relying on. Something changed with 1.16 such that the ipfs binary is no longer found in the PATH. I spent a few hours trying to find where, unsuccessfully. I couldn't find any rationale for MAKE_SKIP_PATH and the tests pass when I disable it, b/c PATH is setup correctly before each test run (pointing to the coverage binary).

Reply:
I'm not totally sure, but I think the idea here is that you can locally run sharness tests on your machine from the sharness test folder without causing weird issues. Could you double check those work.

It seems like removing that flag shouldn't change anything (we set it to 1 in the makefile and then check if it equals 1).

> I'm not sure exactly why this fixes it.

What was failing before and with which code changes? You've also changed a couple commands here (e.g. from ```BIN=$(cd .. && echo `pwd`/bin)``` to `BIN=$(cd .. && pwd)/bin` ). Any chance those were the changes that fixed things and not `MAKE_SKIP_PATH`?

Reply:
I did figure this out finally. It was due to this:

```
echo "export PATH=$(pwd)/go/bin:$PATH" >> $BASH_ENV
```
([link](https://github.com/ipfs/go-ipfs/blob/d3468407aaddbed885abe878f80072186c33a449/.circleci/config.yml#L133))

That writes a PATH env var to BASH_ENV, and BASH_ENV is inherited by every subshell, so every time a subshell is run, Bash applies env vars in BASH_ENV which overwrite whatever PATH was inherited from the parent process, which means setting PATH in any Makefile (or any other env in BASH_ENV) is silently ignored. 

(Setting PATH with BASH_ENV is what CircleCI docs recommend....)